x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/type/DataTypeConverter.java

```diff
@@ -116,7 +121,7 @@ public static boolean canConvert(DataType from, DataType to) {
      */
     public static Converter converterFor(DataType from, DataType to) {
         // Special handling for nulls and if conversion is not requires
-        if (from == to) {
+        if (from == to || (DataTypes.isDateTime(from) && DataTypes.isDateTime(to))) {
```

Comment:
Because currently, DATETIME & DATETIME_NANOS are not the same (internally) but this should be reverted with an upcoming PR that merges them, thus eliminating the DATETIME_NANOS.